app/controllers/bookings_controller.rb

```diff
@@ -1,2 +1,15 @@
 class BookingsController < ApplicationController
+  def create
+    @product = Product.find(params[:product_id])
+    @booking = Booking.new(booking_params)
+    @booking.product = @product
+    @booking.user = current_user
+    @booking.charity = current_user.charity
+    @booking.save
+    redirect_to bookings_path
+  end
+
+  def booking_params
+    params.require(:booking).permit(:status)
+  end
```

Comment:
I think we perhaps don't need this, as there will be no form, and that the status will have a default status (we should all discuss this).